### PR TITLE
Add method to call forgot password lambda

### PR DESCRIPTION
**Pull request by b123400**
connect https://github.com/SkygearIO/skygear-SDK-iOS/issues/92

### CODE REVIEW

**Comment by oursky-travis** [comment]
@b123400, thanks for your PR! By analyzing the history of the files in this pull request, we identified @cheungpat to be a potential reviewer.

File: Pod/Extensions/Forgot Password/SKYContainer+ForgotPassword.m [hunk 1]
```diff
@@ -0,0 +1,41 @@
+//
+//  SKYContainer+Facebook.m
```
Comment (cheungpat) [inline Pod/Extensions/Forgot Password/SKYContainer+ForgotPassword.m]: ditto

File: README.md [hunk 2]
```diff
@@ -49,6 +49,12 @@ For example, if you need facebook login, include the following
 pod 'SKYKit/Facebook'
 ```
 
+For forgot password integration, you can include the following
+
+```
+pod 'SKYKit/ForgotPassword'
```
Comment (cheungpat) [inline README.md]: I think the the `SKYKit/ForgotPassword` subspec should be included in the `SKYKit` spec by default. The README here should be updated to reflect this.

File: SKYKit.podspec [hunk 3]
```diff
@@ -34,4 +34,11 @@ Pod::Spec.new do |s|
     facebook.dependency 'SKYKit/Core'
     facebook.dependency 'FBSDKCoreKit', '~> 4.0'
   end
+
+  s.subspec 'ForgotPassword' do |forgotPassword|
```
Comment (cheungpat) [inline SKYKit.podspec]: * The `default_subspec` should be a new subspec called `Default` (something like that)
  * If user install `SKYKit` it is the same as installing `SKYKit/Default`
* The `Default` subspec should include `Facebook` and `ForgotPassword` subspecs as dependency.

This should be easy. Let me know if there is a problem.

File: Pod/Extensions/Forgot Password/SKYContainer+ForgotPassword.h [hunk 4]
```diff
@@ -0,0 +1,31 @@
+//
+//  SKYContainer+Facebook.h
```
Comment (b123400) [inline Pod/Extensions/Forgot Password/SKYContainer+ForgotPassword.h]: This is so stupid, updated 🙈

**Comment by b123400** [comment]
Updated. The default behaviour of podspec is to include every subspecs, so I think we don't need an extra `Default` subspec.

It is now like this:

```
# Everything
pod 'SKYKit'

# Only a part of the repo
pod 'SKYKit/Core'
pod 'SKYKit/Facebook'
pod 'SKYKit/ForgotPassword'
```

File: Pod/Extensions/Forgot Password/SKYContainer+ForgotPassword.h (hunk 4 above)
Comment (cheungpat) [inline Pod/Extensions/Forgot Password/SKYContainer+ForgotPassword.h]: 👍 Yea I fail the same thing quite a few times.

**Comment by cheungpat** [comment]
> Updated. The default behaviour of podspec is to include every subspecs, so I think we don't need an extra Default subspec.

Oh you can do that? This is even better. Thanks.

File: Pod/Extensions/Forgot Password/SKYContainer+ForgotPassword.m [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+                arguments:@[ emailAddress ]
+        completionHandler:completionHandler];
+}
+
+- (void)resetPasswordWithUserID:(NSString *)userID
+                           code:(NSString *)code
+                       password:(NSString *)password
+              completionHandler:(void (^)(NSDictionary *, NSError *))completionHandler
+{
+    [self callLambda:@"user:reset-password"
```
Comment (cheungpat) [inline Pod/Extensions/Forgot Password/SKYContainer+ForgotPassword.m]: The `user:reset-password` lambda function argument is changing: https://github.com/SkygearIO/forgot_password/pull/16/files#diff-5d586952e502d553206198500bcdf043R176

**Comment by cheungpat** [comment]
Closing this. The author for #110 probably should create a new pull request.

**Pull request closed without merging**